tests/ui/traits/const-traits/const-auto-trait.stderr

```diff
@@ -0,0 +1,10 @@
+error: auto traits cannot be const
+  --> $DIR/const-auto-trait.rs:3:1
+   |
+LL | const auto trait Marker {}
+   | ^^^^^
+   |
+   = help: remove the `const` keyword
```

Comment:
We could provide a structured suggestion but yeah, it's not really worth the effort.

Reply:
Ah, the next-gen solver still crashes despite this change? Right, of course.

I guess we should replace the `unreachable!(…)` in `consider_auto_trait_candidate` of `HostEffectPredicate` with a `cx.delay_bug`. Sorry, I forgot about that.